core/server/master/src/test/java/alluxio/master/file/FileSystemMasterSyncMetadataMetricsTest.java

```diff
@@ -0,0 +1,506 @@
+            false, true, false, false);
+    inodeSyncStream.sync();
+    assertEquals(1 + 1, count.getCount());
+    assertEquals(1 + 1, success.getCount());
+    assertEquals(3 + 1, pathSuccess.getCount());
+    assertEquals(0 + 0, fail.getCount());
+    assertEquals(0 + 0, pathFail.getCount());
+    assertEquals(3 + 0, noChange.getCount());
+    assertEquals(0 + 0, skipped.getCount());
+
+    //sync "/dir0/file0" again
+    //path is existent in inodeTree and UFS and UFS is available ->success and on change
+    syncScheme =
+        new LockingScheme(new AlluxioURI("/dir0/file0"), InodeTree.LockPattern.READ, true);
+    inodeSyncStream =
+        new InodeSyncStream(syncScheme, mFileSystemMaster, RpcContext.NOOP,
+            DescendantType.NONE, ListStatusContext.defaults().getOptions().getCommonOptions(),
+            false, true, false, false);
+    inodeSyncStream.sync();
+    assertEquals(2 + 1, count.getCount());
+    assertEquals(2 + 1, success.getCount());
+    assertEquals(4 + 1, pathSuccess.getCount());
+    assertEquals(0 + 0, fail.getCount());
+    assertEquals(0 + 0, pathFail.getCount());
+    assertEquals(3 + 1, noChange.getCount());
+    assertEquals(0 + 0, skipped.getCount());
+
+    //!forceSync && !shouldSync ->skip
+    syncScheme =
+        new LockingScheme(new AlluxioURI("/dir0/file0"), InodeTree.LockPattern.READ, false);
+    inodeSyncStream =
+        new InodeSyncStream(syncScheme, mFileSystemMaster, RpcContext.NOOP,
+            DescendantType.NONE, ListStatusContext.defaults().getOptions().getCommonOptions(),
+            false,
+            false, false, false);
+    inodeSyncStream.sync();
+    assertEquals(3 + 1, count.getCount());
+    assertEquals(3 + 0, success.getCount());
+    assertEquals(5 + 0, pathSuccess.getCount());
+    assertEquals(0 + 0, fail.getCount());
+    assertEquals(0 + 0, pathFail.getCount());
+    assertEquals(4 + 0, noChange.getCount());
+    assertEquals(0 + 1, skipped.getCount());
+
+    //path is existent in inodeTree and UFS and UFS throws IOException -> success
+    mUfs.mIsThrowIOException = true;
+    syncScheme = new LockingScheme(new AlluxioURI("/dir0/file0"), InodeTree.LockPattern.READ, true);
+    inodeSyncStream = new InodeSyncStream(syncScheme, mFileSystemMaster, RpcContext.NOOP,
+        DescendantType.NONE, ListStatusContext.defaults().getOptions().getCommonOptions(),
+            false, true, false, false);
+    inodeSyncStream.sync();
+    assertFalse(mFileSystemMaster.getInodeTree().inodePathExists(new AlluxioURI("/dir0/file0")));
+    assertEquals(4 + 1, count.getCount());
+    assertEquals(3 + 1, success.getCount());
+    assertEquals(5 + 1, pathSuccess.getCount());
+    assertEquals(0 + 0, fail.getCount());
+    assertEquals(0 + 0, pathFail.getCount());
+    assertEquals(4 + 0, noChange.getCount());
+    assertEquals(1 + 0, skipped.getCount());
+
+    //path is non-existent in inodeTree,existent in UFS and UFS throws IOException -> fail
```

Comment:
```suggestion
    // The path is non-existent in inodeTree and existent in UFS
    // The sync should fail because UFS throws IOException
```

Reply:
pls double check the rest of comments and try to improve the expression

Reply:
done